src/optimizer.cpp

```diff
@@ -84,14 +97,14 @@ void Optimizer::setOffset(double controller_frequency)
   const double controller_period = 1.0 / controller_frequency;
   constexpr double eps = 1e-6;
 
-  if (controller_period < settings_.model_dt) {
+  if ((controller_period + eps) < settings_.model_dt) {
```

Comment:
Note the bug fix, otherwise the else if wouldn't be called if there is some small numerical difference making the control period lower due to the inversion a few lines up